File: pset1/greedy.c
#include <stdio.h>

#include <cs50.h>

#include <math.h>

int main(void)

{
    float dlr, cents;
    
    int counter=0;
    
    do
    {
        printf ("Enter the amount\n");
        
       dlr= GetFloat ();
        
    }
    
    while (dlr<0);                          //Protection from non float input.

cents=round(dlr*100);                       //converting dollars into cents and rounding off float.

while (cents>=25)
{
cents=cents-25;

counter++;                                  //For counting loop cycles.
}

while (cents>=10)
{
    cents=cents-10;
    counter++;
}

while (cents>=5)
{
    cents=cents-5;
    counter++;
}

while (cents>=1)
{
    cents=cents-1;
    counter++;
}

printf("%d\n", counter);

}

File: pset6/greedy.py
import cs50

while (True):
    print("please give positive input")
    dlr = cs50.get_float()
    if dlr > 0:
        break
cents = round (dlr*100)
counter = 0
while (cents >= 25):
    cents -= 25
    counter += 1
    
while (cents >= 10):
    cents -= 10
    counter += 1
   
while (cents >= 5):
    cents -= 5
    counter += 1
    
while (cents > 0):
    cents -= 1
    counter += 1    
   
print (counter)